settings.coffee

```diff
@@ -213,7 +213,7 @@ settings =
 		collaborators: -1
 		dropbox: true
 		versioning: true
-		compileTimeout: process.env["COMPILE_TIMEOUT"] or 180
+		compileTimeout: parseInt(process.env["COMPILE_TIMEOUT"]) or 180
```

Comment:
Actually... it would be [good practice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt) to specify the radix here. And I think there's an argument for writing it as
```js
parseInt(process.env["COMPILE_TIMEOUT"] or 180, 10)
```
so that if the provided compile timeout is not a valid number, it doesn't silently default to 180 (since `NaN` is falsy).

Reply:
Updated in https://github.com/overleaf/overleaf/pull/805/commits/d8d6dbffee981d3966db270ea76687548e92bfb4, also added `radix` in the other `parseInt()` calls in https://github.com/overleaf/overleaf/pull/806.

> if the provided compile timeout is not a valid number, it doesn't silently default to 180

I'm not so sure this is the best behaviour for invalid numbers. Having a `compileTimeout` as `NaN` might be a bit unpredictable or obscure to fail.

What do you think about adding a custom `parseInt()` function that throws an error when the string to parse is invalid? That would be the shortest feedback loop for incorrect settings.

Reply:
Good point. I think that would be a useful helper to have 👍 . I agree that ideally we should be validating these env vars and failing fast.

Reply:
Thanks, I'll merge this as it is, and will make https://github.com/overleaf/overleaf/pull/806 fail fast. 

Ideally it would be nice to have the fast-fail for all the settings across modules/products, but that's particularly valuable for Server Pro customers, so I think it's ok to start with this settings file.